Assets/Scripts/InteractionSystem/IslandHeartInteractBase.cs

```diff
@@ -0,0 +1,128 @@
+using UnityEngine;
+using Player.View;
+using Player.Motion;
+using TMPro;
+using UnityEngine.UI;
+using Jellies;
+using UI.Inventory;
+
+public class IslandHeartInteractBase : Interactable
+{
+    [Header("Timing")]
+    [SerializeField]
+    private float _reactivateTime = 1.25f;
+
+    [Header("Controlled Objects")]
+    [SerializeField, Tooltip("Virtual Island Heart camera to focus for interaction.")]
+    private GameObject _islandHeartVCamera;
+
+    [Header("Feed-Island Heart interaction UI")]
+    [SerializeField, Tooltip("If the Island Heart is leveling up, disable the feedButton")]
+    private Button _feedButton;
+
+    private IslandHeartLeveling heart;
+
+    private GameObject _interactionUI;
+    private FirstPersonView _firstPersonView;
+    private PlayerInteraction _playerInteraction;
+
+
+    private bool _interacting;
+    public bool Interacting
+    {
+        get => _interacting;
+        private set
+        {
+            if (value != _interacting)
+            {
+                _interacting = value;
+
+                // Only 1 can interact at a time, so we can determine whether
+                // any are interacting like this. Need the static one, but also
+                // need to know whether this specific one is interacting.
+                AnyInteracting = value;
+            }
+        }
+    }
+    public static bool AnyInteracting { get; private set; }
```

Comment:
This is fine for now.

When we sort out the interaction system, we'll need to get rid of this duplicate code. JellyInteractBase and this both inherit Interactable but don't seem to use it. They have similar code too. 

I'm not sure we even need to use inheritance for interactables. I feel like it's being used for too much. We'd need to split out some things, e.g. the jelly's saturation bar could be updated from a different monobehaviour.